browser-extension/mv3/src/page-scripts/ajaxRequestInterceptor/utils.js

```diff
@@ -1,12 +1,12 @@
-import { PUBLIC_NAMESPACE } from "common/constants";
+import { EXTENSION_MESSAGES, PUBLIC_NAMESPACE } from "common/constants";
 import { matchRuleWithRequest, matchSourceUrl } from "../../common/ruleMatcher";
 import { SourceKey, SourceOperator } from "common/types";
 
 let logShown = false;
 
 export const getFunctionFromCode = (code, ruleType) => {
   try {
-    return new Function("args", `return (${code})(args);`);
+    return new Function("args,sharedState", `return (${code})(args,sharedState);`);
```

Comment:
@wrongsahil @nafees87n I think now would be a good time to implement the same approach in both proxy and extension for how we handle function code strings.

because in the proxy we do it slightly differently _(following is before the [sharedState changes](https://github.com/requestly/requestly-proxy/blob/12354850a84817a73ea27d9c0ede472e8f10cc1b/src/utils/index.ts#L5-L7))_:

```typescript
export const getFunctionFromString = function (functionStringEscaped) {
    return new Function("return " + functionStringEscaped)();
};
```

This generates the function that is then called as `userFunction(args)`

> The difference is where the scope of `args` is. Its trivial but good to keep it consistent across both the implementations

Reply:
@nafees87n please refer the changes in [this file](https://github.com/requestly/requestly-proxy/blob/master/src/utils/index.ts) for the desktop implementation